src/main/java/de/zalando/aruha/nakadi/service/subscription/state/ClosingState.java

```diff
@@ -50,6 +50,9 @@ public void onEnter() {
     }
 
     private void onTopologyChanged() {
+        if (topologyListener == null) {
+            throw new IllegalStateException("topologyListener should not be null when calling onTopologyChanged method");
```

Comment:
I mean, isn't it possible to "fail early". If there is any problem when creating the listener, the error should be thrown in that moment and not afterwards when trying to use it. Makes sense?